File: leetcode/238. Product of Array Except Self.ts
//https://leetcode.com/problems/product-of-array-except-self/description/
function productExceptSelf(nums: number[]): number[] {
    const res: number[] = []

    let acc: number = 1
    for(let i = 0; i < nums.length; i++) {
        res.push(acc)
        acc *= nums[i]
    }

    acc = 1
    for(let i = nums.length-1; i >= 0; i--) {
        res[i] *= acc
        acc *= nums[i]
    }

    return res
};

File: leetcode/238. Product of Array Except Self.go
//https://leetcode.com/problems/product-of-array-except-self/
func productExceptSelf(nums []int) []int {
    res := []int{}

    start := 1
    for _, v := range nums {
        res = append(res, start)
        start = v * start
        fmt.Println("d", start)
    }

    //end
    start = 1
    for i := len(nums)-1; i >= 0; i-- {
        res[i] = start * res[i]
        start= start*nums[i]
    }

    return res
}